File: backend/schema/plant.py
from pydantic import BaseModel
from typing import List

from backend.schema.activeCompound import ActiveCompound


class Plant(BaseModel):
    id: int
    scientificName: str
    author: str
    familyName: str
    localName: str
    utilizedPart: str
    ailment: str
    activeCompound: str
    pmid: str
    pmAcList: List[ActiveCompound]

File: frontend/src/entity/Plant.ts
import ActiveCompound from "@/entity/ActiveCompound";

class Plant{
    constructor(
        public id: number,
        public scientificName: string,
        public author: string,
        public familyName: string,
        public localName: string,
        public ailment: string,
        public activeCompound: string,
        public pmid: string,
        public utilizedPart: string,
        public pmAcList: Array<ActiveCompound>
    ) {}

}

export default Plant;